operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/DefaultResourceConfiguration.java

```diff
@@ -4,46 +4,38 @@
 import java.util.Set;
 
 import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.ReconcilerUtils;
 import io.javaoperatorsdk.operator.processing.event.source.filter.GenericFilter;
 import io.javaoperatorsdk.operator.processing.event.source.filter.OnAddFilter;
 import io.javaoperatorsdk.operator.processing.event.source.filter.OnUpdateFilter;
 
-import static io.javaoperatorsdk.operator.api.reconciler.Constants.DEFAULT_NAMESPACES_SET;
-
 public class DefaultResourceConfiguration<R extends HasMetadata>
     implements ResourceConfiguration<R> {
 
+  private final Class<R> resourceClass;
+  private final String resourceTypeName;
+  private final Optional<OnAddFilter<R>> onAddFilter;
```

Comment:
Why these Optionals? It's kinda not common practice to use it here. I would rather stick with the former version.

Reply:
I can go either way. The idea here was to resolve the optional at creation instead of creating a new instance each time the getter is called.

Reply:
Would rather stick to the convention tbh

Reply:
Removed.